src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java

```diff
@@ -64,9 +70,50 @@ public TestdataStatement()
     public void evaluate() throws Throwable
     {
         Neodymium.getData().putAll(testData);
+        initializeDataObjects();
         next.evaluate();
     }
 
+    private void initializeDataObjects() throws IllegalArgumentException, IllegalAccessException
+    {
+        for (Field field :getFieldsFromSuperclasses())
+        {
+            Data dataAnnotation = field.getAnnotation(Data.class);
+            if (dataAnnotation != null)
+            {
+                boolean isFieldAccessable = field.canAccess(testClassInstance);
+                field.setAccessible(true);
+                try
```

Comment:
I tested the feature and would like to provide an easy access to the first level of objects. Please adjust the code and the test cases accordingly to the following snippet.

```
                try
                {
                    if (!StringUtils.isBlank(dataAnnotation.value()))
                    {
                        field.set(testClassInstance, DataUtils.get(dataAnnotation.value(), field.getType()));
                    }
                    else
                    {
                        if (DataUtils.exists(field.getName()))
                        {
                            field.set(testClassInstance, DataUtils.get("$." + field.getName(), field.getType()));
                        }
                        else
                        {
                            field.set(testClassInstance, DataUtils.get(field.getType()));
                        }
                    }
                } catch (Exception e) {
                    // TODO: throw a RuntimeException that contains a hint how this  error can be taken care of
                }
                finally
                {
                    field.setAccessible(isFieldAccessable);
                }
```

Reply:
We will also need new tests regarding the first level object support.

Reply:
Done. I haven't created a new test for it as I think, that the `InstantiateFieldViaAnnotation` is suitable for this. So I adjusted this test to use field name instead of JSON path. The JSON path is already tested in the `InstantiateDtoViaJsonPathInAnnotation` anyway.